### PR TITLE
Lesson3_Task1

### PR DESCRIPTION
**Pull request by stalgatuly**
Lesson3_Task1:
1) Find the average of an array
2) Find the longest word in a slice
3) Reverse a slice
4) Print the map values sorted in order of increasing keys

### CODE REVIEW

**Review by nikonok — APPROVED** [review]

File: lesson_03/task03_1.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+func avgArray(arr [10]int) float64 {
+	var total float64 = 0
+	for _, v := range arr {
+		total += float64(v)
+	}
+	return float64(total / 10)
+}
+
+func max(s []string) string {
```
Comment (denislituev) [inline lesson_03/task03_1.go]: It is better to give more sensible namings, brcause it is not clear what is 'max', it's better 'getLongestString' or smth like this

File: lesson_03/task03_1.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	if len(s) > 0 {
+		maxlen, answerIndex := utf8.RuneCountInString(s[0]), 0
+		for i, v := range s {
+			if utf8.RuneCountInString(v) > maxlen {
+				maxlen, answerIndex = utf8.RuneCountInString(v), i
+			}
+		}
+		return s[answerIndex]
+	} else {
```
Comment (denislituev) [inline lesson_03/task03_1.go]: It's better to check for error cases or do some checks for empty args in the beginning and return error or empty result at once. This approach wil be more as Go way :)

In this current function we should check for len(s) == 0 and return "" at once, and then we don't need to use 'else'

File: lesson_03/task03_1.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+		return ""
+	}
+}
+
+func reverse(s []int64) []int64 {
+	newslice := []int64{}
```
Comment (denislituev) [inline lesson_03/task03_1.go]: here we can create newslice with len(s) using make, to avoid slice extension, which could be several times

**Review by denislituev — APPROVED** [review]
Approved with suggestions. You can fix this comments or merge without fixing, but I want you to look through it and try to remember it

**Pull request merged**